packages/activemodel-adapter/lib/system/active-model-serializer.js

```diff
@@ -166,7 +166,9 @@ var ActiveModelSerializer = RESTSerializer.extend({
     if (Ember.isNone(belongsTo)) {
       json[jsonKey] = null;
     } else {
-      json[jsonKey] = capitalize(camelize(belongsTo.typeKey));
+      json[jsonKey] = classify(belongsTo.typeKey).replace(/(\/)([a-z])/g, function(match, separator, chr) {
```

Comment:
Perhaps this could be rewritten as `classify(belongsTo.typeKey).split('/').map(classify).join('::')` ?